File: BackOffice.Presentation/src/types/auditLog.ts
export interface AuditLogGridItem {
  id: number;
  userId: number | null;
  action: string;
  entityType: string;
  entityId: string | null;
  changedFields: string | null;
  createdAt: string;
}

export interface AuditLogDetail {
  id: number;
  userId: number | null;
  action: string;
  entityType: string;
  entityId: string | null;
  oldValue: string | null;
  newValue: string | null;
  changedFields: string | null;
  ipAddress: string | null;
  createdAt: string;
}

export interface FieldChange {
  field: string;
  oldValue: string | null;
  newValue: string | null;
}

File: BackOffice.Domain/Entities/Tenant/AuditLog.cs
#nullable enable
using System;
using BackOffice.Domain.Attributes;

namespace BackOffice.Domain.Entities.Tenant;

[NotAudited]
public partial class AuditLog
{
    public long Id { get; set; }

    public int? UserId { get; set; }

    public string Action { get; set; } = null!;

    public string EntityType { get; set; } = null!;

    public string? EntityId { get; set; }

    public string? OldValue { get; set; }

    public string? NewValue { get; set; }

    public string? ChangedFields { get; set; }

    public string? IpAddress { get; set; }

    public DateTime CreatedAt { get; set; }
}
